### PR TITLE
session: add GoString() for TxnState and make the log more use friendly

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

Make some log more user friendly

### What is changed and how it works?

Implement  `GoString()` for `TxnState` so the fmt.Printf("%#v") will work.
The fmt.Println("%v") format for `TxnState` is terrible .

### Check List 

Tests 

 - Unit test

PTAL @jackysp

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8434)

### CODE REVIEW

**Comment by jackysp** [comment]
Please fix `make check`.

File: session/txn.go [hunk 1]
```diff
@@ -77,6 +80,33 @@ func (st *TxnState) String() string {
...
+	if st.pending() {
+		s.WriteString("state=pending")
+	} else if st.Valid() {
+		s.WriteString("state=valid")
+		fmt.Fprintf(&s, ", startTS=%d", st.Transaction.StartTS())
+		if len(st.dirtyTableOP) > 0 {
+			fmt.Fprintf(&s, ", len(dirtyTable)=%d", len(st.dirtyTableOP))
+		}
+		if len(st.mutations) > 0 {
+			fmt.Fprintf(&s, ", len(mutations)=%d", len(st.mutations))
```
Comment (lysu) [inline session/txn.go]: I think print `st.mutations`'s content is useful, we can see `kind, tid, handle int64` but this PR miss them???

Comment (tiancaiamao) [reply session/txn.go]: So this is a dilemma:
when logs too much, you will think they are totally a mess
when logs too little, you find it hard to debug

Comment (winkyao) [reply session/txn.go]: print len(mutations) is useless?

Comment (tiancaiamao) [reply session/txn.go]: `len(st.mutations) > 0` we know how many things are changed.
If we print mutation data, it maybe too much. @winkyao

**Pull request merged**